src/effects/EffectBase.h

```diff
@@ -37,10 +38,17 @@ class AUDACITY_DLL_API EffectBase /* not final */
 
    void SetTracks(TrackList *pTracks) { mTracks = pTracks; }
 
+   //! Called when Preview() starts, to allow temporary effect state changes
+   /*!
+    default returns a null
+    @return will undo its effects in its destructor before Preview() finishes
+    */
+   virtual std::any BeginPreview(const EffectSettings &settings);
```

Comment:
If it is called only by Preview, why is it public? It feels that it should be protected instead.

Reply:
Actually, this is a rare case where I don't disagree with `any` :-)

Reply:
Protected makes sense.

std::any is just type erasure for things that can copy and may have nontrivial destructors.

My type erased pipeline class also uses it.